### ISSUE TITLE
Error: EBADF: bad file descriptor, (node v21.3.0)

### ISSUE THREAD

**Issue by padcom** [qmhc/vite-plugin-dts#288]
### Describe the bug

When using this plugin with Node.js v21.3.0 the following error occurs:

```
Error: EBADF: bad file descriptor, write
    at Object.writeFileSync (node:fs:2352:20)
    at /home/padcom/projects/@padcom/vuelit/node_modules/@rushstack/node-core-library/lib/FileSystem.js:466:21
    at FileSystem._wrapException (/home/padcom/projects/@padcom/vuelit/node_modules/@rushstack/node-core-library/lib/FileSystem.js:1050:20)
    at FileSystem.writeFile (/home/padcom/projects/@padcom/vuelit/node_modules/@rushstack/node-core-library/lib/FileSystem.js:460:20)
    at DtsRollupGenerator.writeTypingsFile (/home/padcom/projects/@padcom/vuelit/node_modules/@microsoft/api-extractor/lib/generators/DtsRollupGenerator.js:81:40)
    at Extractor._generateRollupDtsFile (/home/padcom/projects/@padcom/vuelit/node_modules/@microsoft/api-extractor/lib/api/Extractor.js:288:53)
    at Extractor.invoke (/home/padcom/projects/@padcom/vuelit/node_modules/@microsoft/api-extractor/lib/api/Extractor.js:230:23)
    at rollupDeclarationFiles (file:///home/padcom/projects/@padcom/vuelit/node_modules/vite-plugin-dts/dist/index.mjs:272:28)
    at Object.writeBundle (file:///home/padcom/projects/@padcom/vuelit/node_modules/vite-plugin-dts/dist/index.mjs:933:13)
    at async Promise.all (index 0)
```

### Reproduction

https://github.com/padcom/vite-dts-plugin-error-with-node-v21

### Steps to reproduce

npm install
npm run build

### System Info

```shell
System:
    OS: Linux 6.2 Linux Mint 21.2 (Victoria)
    CPU: (24) x64 13th Gen Intel(R) Core(TM) i7-13700K
    Memory: 18.95 GB / 31.10 GB
    Container: Yes
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 21.3.0 - ~/.nvm/versions/node/v21.3.0/bin/node
    npm: 10.2.4 - ~/.nvm/versions/node/v21.3.0/bin/npm
  Browsers:
    Chrome: 119.0.6045.159
    Chromium: 119.0.6045.159
  npmPackages:
    vite: ^5.0.4 => 5.0.4 
    vite-plugin-dts: ^3.6.4 => 3.6.4
```

### Validations

- [X] Read the [FAQ](https://github.com/qmhc/vite-plugin-dts#faq).
- [X] Check that there isn't [already an issue](https://github.com/qmhc/vite-plugin-dts/issues) that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by padcom** [comment]
Please note version node.js v20.10.0 works without issues.

**Comment by thenick775** [comment]
➕

**Comment by qmhc** [comment]
Is this issue occurring if the version of node.js large than `21.3`?

**Comment by qmhc** [comment]
This is a bug of node.js, it's fixed by https://github.com/nodejs/node/issues/50989 and it's released in 21.4.0.
